### ISSUE TITLE
EAS Submit failing on resolving plugins

### ISSUE THREAD

**Issue by tidusjar** [expo/expo#23284]
### Summary

When submitting a build via `eas submit` for both android and ios builds I'm getting the following error:
NOTE: this worked fine in Expo 47

```
Failed to resolve plugin for module "sentry-expo" relative to "/home/runner/work/ReactNativeOmbi/ReactNativeOmbi"
    Error: submit command failed.
Error: Process completed with exit code 1.
```

### What platform(s) does this occur on?

Android, iOS

### SDK Version

48.0.0

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 16.13.2 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.11 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ^48.0.0 => 48.0.19
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.71.8 => 0.71.8
      react-native-web: ~0.18.7 => 0.18.12
    Expo Workflow: managed
```

### Minimal reproducible example

package.json

```
{
  "main": "node_modules/expo/AppEntry.js",
  "version": "2.15.0",
  "jest": {
    "preset": "jest-expo",
    "setupFiles": [
      "./jest.setup.js"
    ],
    "transformIgnorePatterns": [
      "node_modules/(?!(jest-)?react-native|react-clone-referenced-element|@react-native-community|expo(nent)?|@expo(nent)?/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|@sentry/.*)"
    ],
    "setupFilesAfterEnv": [
      "@testing-library/jest-native/extend-expect"
    ]
  },
  "scripts": {
    "version": "npx standard-version",
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "lint": "eslint --ext .tsx --ext .ts src/",
    "test": "jest",
    "build:and": "expo build:android -t app-bundle --release-channel prod-2.8.0",
    "build:and:apk": "expo build:android -t apk --release-channel prod-2.8.0",
    "build:ios": "expo build:ios -t archive --release-channel prod-2.8.0 ",
    "upload:and:latest": "expo upload:android --latest --key ./tools/api-7777968725427313635-951042-0330a56d81c7.json --track beta"
  },
  "dependencies": {
    "@eva-design/eva": "^2.1.1",
    "@react-native-async-storage/async-storage": "1.17.11",
    "@react-native-community/masked-view": "0.1.10",
    "@react-native-community/slider": "4.4.2",
    "@react-navigation/bottom-tabs": "^5.11.11",
    "@react-navigation/native": "^5.9.3",
    "@react-navigation/stack": "^5.14.5",
    "@shopify/flash-list": "1.4.0",
    "@types/react-native-snap-carousel": "^3.8.4",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "@ui-kitten/components": "^5.1.2",
    "@ui-kitten/eva-icons": "^5.1.2",
    "awesome-debounce-promise": "^2.1.0",
    "axios": "^1.3.2",
    "date-fns": "^2.29.3",
    "eas-cli": "^3.5.1",
    "eslint": "^7.22.0",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.23.1",
    "eslint-plugin-react-hooks": "^4.2.0",
    "expo": "^48.0.0",
    "expo-application": "~5.1.1",
    "expo-barcode-scanner": "~12.3.2",
    "expo-constants": "~14.2.1",
    "expo-device": "~5.2.1",
    "expo-haptics": "~12.2.1",
    "expo-image": "~1.0.1",
    "expo-linking": "~4.0.1",
    "expo-localization": "~14.1.1",
    "expo-notifications": "~0.18.1",
    "expo-splash-screen": "~0.18.2",
    "expo-status-bar": "~1.4.3",
    "expo-store-review": "~6.2.1",
    "expo-system-ui": "~2.2.1",
    "expo-updates": "~0.16.4",
    "formik": "^2.2.9",
    "global": "^4.4.0",
    "i18n-js": "^3.9.2",
    "lodash": "^4.17.21",
    "mobx": "^6.7.0",
    "mobx-react-lite": "^3.4.0",
    "react": "18.2.0",
    "react-async-hook": "^4.0.0",
    "react-dom": "18.2.0",
    "react-native": "0.71.8",
    "react-native-draggable-flatlist": "^4.0.0",
    "react-native-dropdownalert": "^4.5.1",
    "react-native-elements": "^3.4.3",
    "react-native-fade-in-image": "^1.6.1",
    "react-native-gesture-handler": "~2.9.0",
    "react-native-modalize": "^2.1.1",
    "react-native-read-more-text": "^1.1.2",
    "react-native-responsive-dimensions": "^3.1.1",
    "react-native-safe-area-context": "4.5.0",
    "react-native-screens": "~3.20.0",
    "react-native-skeleton-placeholder": "^3.0.4",
    "react-native-snap-carousel": "3.9.1",
    "react-native-svg": "13.4.0",
    "react-native-web": "~0.18.7",
    "react-native-webview": "11.26.0",
    "react-query": "^3.39.3",
    "sentry-expo": "~6.2.0",
    "use-constant": "^1.1.0",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@testing-library/jest-native": "^5.4.1",
    "@testing-library/react-native": "^11.5.1",
    "@types/i18n-js": "^3.8.4",
    "@types/jest": "^29.4.0",
    "@types/lodash": "^4.14.191",
    "@types/react-native-read-more-text": "^1.0.1",
    "@types/react-test-renderer": "^17.0.1",
    "jest": "^29.2.1",
    "jest-expo": "^47.0.0",
    "react-native-svg-transformer": "^1.0.0",
    "standard-version": "^9.3.1",
    "standard-version-expo": "^1.0.3",
    "typescript": "^4.6.3"
  },
  "private": true
}
```

app.json plugins

```
    "plugins": [
      "sentry-expo"
    ]
```

**Comment by RRaideRR** [comment]
far from being an expert here, but the [documentation](https://docs.expo.dev/guides/using-sentry/) says that you should also install `@sentry/react-native` which I cannot see in your package.json

**Comment by tidusjar** [comment]
Still no luck, excatly the same erorr even with the dependency included

**Comment by brentvatne** [comment]
can you run this and share the output?

`DEBUG=* eas submit`

**Comment by tidusjar** [comment]
Output:

```
2023-0[7](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:8)-04T20:12:47.475Z config reading core plugin /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli
2023-07-04T20:12:47.476Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli/package.json
2023-07-04T20:12:47.479Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli/oclif.manifest.json
2023-07-04T20:12:47.4[8](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:9)0Z config:eas-cli using manifest from /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli/oclif.manifest.json
2023-07-04T20:12:47.48[9](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:10)Z config reading user plugins pjson /home/runner/.local/share/eas-cli/package.json
2023-07-04T20:12:47.489Z config loadJSON /home/runner/.local/share/eas-cli/package.json
2023-07-04T20:12:47.489Z config loading plugins [
  '@oclif/plugin-autocomplete',
  '@expo/plugin-help',
  '@expo/plugin-warn-if-update-available'
]
2023-07-04T20:12:47.492Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli/package.json
2023-07-04T20:12:47.493Z config reading core plugin /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-help
2023-07-04T20:12:47.493Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-help/package.json
2023-07-04T20:12:47.493Z config reading core plugin /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-warn-if-update-available
2023-07-04T20:12:47.494Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-warn-if-update-available/package.json
2023-07-04T20:12:47.495Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/package.json
2023-07-04T20:12:47.495Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-help/oclif.manifest.json
2023-07-04T20:12:47.496Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-warn-if-update-available/oclif.manifest.json
2023-07-04T20:12:47.496Z config reading core plugin /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@oclif/plugin-autocomplete
2023-07-04T20:12:47.497Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@oclif/plugin-autocomplete/package.json
2023-07-04T20:12:47.497Z config:@expo/plugin-warn-if-update-available using manifest from /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-warn-if-update-available/oclif.manifest.json
2023-07-04T20:12:47.498Z config:@expo/plugin-help using manifest from /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-help/oclif.manifest.json
2023-07-04T20:12:47.499Z config loadJSON /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@oclif/plugin-autocomplete/oclif.manifest.json
2023-07-04T20:12:47.499Z config:@oclif/plugin-autocomplete using manifest from /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@oclif/plugin-autocomplete/oclif.manifest.json
2023-07-04T20:12:47.501Z config config done
2023-07-04T20:12:47.502Z config start init hook
2023-07-04T20:12:47.529Z eas:@expo/plugin-warn-if-update-available:hooks:init start (require) /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/plugin-warn-if-update-available/lib/hooks/init/check-update.js
2023-07-04T20:12:47.531Z update-check Error: ENOENT: no such file or directory, stat '/home/runner/.cache/eas-cli/version'
2023-07-04T20:12:47.531Z update-check spawning version refresh
2023-07-04T20:12:47.535Z eas:@expo/plugin-warn-if-update-available:hooks:init done
2023-07-04T20:12:47.535Z config init hook done
2023-07-04T20:12:47.536Z config runCommand submit [ '-p', 'android', '--latest' ]
2023-07-04T20:12:48.325Z config:eas-cli (require) /opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/eas-cli/build/commands/submit.js
2023-07-04T20:12:48.326Z config start prerun hook
2023-07-04T20:12:48.326Z config prerun hook done
2023-07-04T20:12:48.327Z eas:submit init version: @oclif/core@1.23.2 argv: [ '-p', 'android', '--latest' ]
Failed to resolve plugin for module "sentry-expo" relative to "/home/runner/work/ReactNativeOmbi/ReactNativeOmbi"
PluginError: Failed to resolve plugin for module "sentry-expo" relative to "/home/runner/work/ReactNativeOmbi/ReactNativeOmbi"
    at resolvePluginForModule (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:98:11)
    at resolveConfigPluginFunctionWithInfo (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:188:7)
    at resolveConfigPluginFunction (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:179:7)
    at withStaticPlugin (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config-plugins/build/plugins/withStaticPlugin.js:[10](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:11)8:70)
    at /opt/hostedtoolcache/eas-cli/3.[15](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:16).0/x64/node_modules/@expo/config-plugins/build/plugins/withPlugins.js:38:84
    at Array.reduce (<anonymous>)
    at withPlugins (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config-plugins/build/plugins/withPlugins.js:38:[18](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:19))
    at withConfigPlugins (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config/build/plugins/withConfigPlugins.js:45:47)
    at fillAndReturnConfig (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config/build/Config.js:[31](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:32)5:78)
    at getConfig (/opt/hostedtoolcache/eas-cli/3.15.0/x64/node_modules/@expo/config/build/Config.js:[38](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5458120472/jobs/9932875468#step:5:39)2:10)
Error: Process completed with exit code 1.
```

**Comment by brentvatne** [comment]
- is your project using a monorepo?
- does `npx expo config --type public` work?
- does `eas config` work?

**Comment by tidusjar** [comment]
It's not a monorepo.

`npx expo config --type public` gives an output

```
{
  scheme: 'ombi',
  name: 'Ombi',
  owner: 'tidusjar',
  slug: 'Ombi',
  version: '2.15.0',
  backgroundColor: '#0f171f',
  icon: './src/assets/icon.png',
  userInterfaceStyle: 'automatic',
  assetBundlePatterns: [
    '**/*'
  ],
  plugins: [
    'sentry-expo'
  ],
  description: undefined,
  sdkVersion: '48.0.0',
  platforms: [
    'ios',
    'android',
    'web'
  ],
  notification: {
    icon: './src/assets/notification/icon.png',
    color: '#62d2fa'
  },
  splash: {
    image: './src/assets/splash.png',
    resizeMode: 'contain',
    backgroundColor: '#0f171f'
  },
  updates: {
    fallbackToCacheTimeout: 0,
    url: 'https://u.expo.dev/622701ac-bb8e-43c1-a0f2-d1d72684b8e8'
  },
  ios: {
    supportsTablet: true,
    appStoreUrl: 'https://apps.apple.com/gb/app/ombi/id1335260043',
    bundleIdentifier: 'com.tidusjar.ombi',
    icon: './src/assets/iosIcon.png',
    buildNumber: '2.15.0',
    infoPlist: {
      NSCameraUsageDescription: 'This app uses the camera to scan barcodes to provide a more user friendly way of logging in.',
      LSApplicationQueriesSchemes: [
        'plex',
        'emby'
      ],
      UIBackgroundModes: [
        'remote-notification'
      ]
    }
  },
  android: {
    versionCode: 480021500,
    package: 'com.tidusjar.Ombi',
    playStoreUrl: 'https://play.google.com/store/apps/details?id=com.tidusjar.Ombi',
    googleServicesFile: './tools/google-services.json',
    permissions: [
      'CAMERA',
      'com.google.android.c2dm.permission.RECEIVE'
    ],
    intentFilters: [
      {
        action: 'MAIN',
        category: [
          'LEANBACK_LAUNCHER',
          'LAUNCHER'
        ],
        data: {}
      }
    ],
    adaptiveIcon: {
      foregroundImage: './src/assets/adaptive-icon.png',
      backgroundColor: '#0f171f'
    }
  },
  web: {
    favicon: './src/assets/favicon.png'
  },
  packagerOpts: {
    config: 'metro.config.js',
    sourceExts: [
      'expo.ts',
      'expo.tsx',
      'expo.js',
      'expo.jsx',
      'ts',
      'tsx',
      'js',
      'jsx',
      'json',
      'wasm',
      'svg'
    ]
  },
  extra: {
    eas: {
      projectId: '622701ac-bb8e-43c1-a0f2-d1d72684b8e8'
    }
  },
  runtimeVersion: {
    policy: 'sdkVersion'
  }
}
```

`eas config` also works


Happy to invite someone to the repo to see

**Comment by tidusjar** [comment]
Interestingly running `eas submit -p android --latest` locally via the terminal works fine, just seems to be via Github Actions it doesn't work

```
# This is a basic workflow to help you get started with Actions

name: Submit to App Store

# Controls when the action will run.
on:
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  Release:
    runs-on: ubuntu-latest
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: 16.x

      - name: Setup Expo
        uses: expo/expo-github-action@v6
        with:
          eas-version: latest
          expo-version: 6.x
          token: ${{ secrets.EXPO_TOKEN }}
          expo-cache: true

      - name: Submit iOS
        run: eas submit -p ios --latest
```

**Comment by brentvatne** [comment]
@tidusjar - try comparing the output of `npx expo config --type public` on your github action to your local output

**Comment by tidusjar** [comment]
I get the following error when attempting to run that as part of the build: 

```
Run npx expo config --type public
npm WARN exec The following package was not found and will be installed: expo@49.0.0
npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
PluginError: Failed to resolve plugin for module "sentry-expo" relative to "/home/runner/work/ReactNativeOmbi/ReactNativeOmbi"
PluginError: Failed to resolve plugin for module "sentry-expo" relative to "/home/runner/work/ReactNativeOmbi/ReactNativeOmbi"
    at resolvePluginForModule (/home/runner/.npm/_npx/249ca9fcd30c47[6](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5484127113/jobs/9991266881#step:5:7)a/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:[7](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5484127113/jobs/9991266881#step:5:8)2:11)
    at resolveConfigPluginFunctionWithInfo (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:152:7)
    at resolveConfigPluginFunction (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js:143:7)
    at withStaticPlugin (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config-plugins/build/plugins/withStaticPlugin.js:[8](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5484127113/jobs/9991266881#step:5:9)7:70)
    at /home/runner/.npm/_npx/24[9](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5484127113/jobs/9991266881#step:5:10)ca9fcd30c476a/node_modules/@expo/config-plugins/build/plugins/withPlugins.js:30:84
    at Array.reduce (<anonymous>)
    at withPlugins (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config-plugins/build/plugins/withPlugins.js:30:18)
    at withConfigPlugins (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config/build/plugins/withConfigPlugins.js:36:47)
    at fillAndReturnConfig (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config/build/Config.js:217:78)
    at getConfig (/home/runner/.npm/_npx/249ca9fcd30c476a/node_modules/@expo/config/build/Config.js:272:[10](https://github.com/tidusjar/ReactNativeOmbi/actions/runs/5484127113/jobs/9991266881#step:5:11))
Error: Process completed with exit code 1.
```

So running a `yarn install` before actually allows this to work now.

Interestingly that was not needed for v47
